src/chat/chatter.py

```diff
@@ -273,47 +272,73 @@ def updatePlayer(self):
         # Rating icon choice  (chr(0xB1) = +-)
         formatting = ("Global Rating: {} ({} Games) [{}\xb1{}]\n"
                       "Ladder Rating: {} [{}\xb1{}]")
-        self.rankItem.setToolTip(formatting.format(
+        tooltip_str = formatting.format(
             str(int(rating)),
             str(player.number_of_games),
             str(int(player.rating_mean)),
             str(int(player.rating_deviation)),
             str(int(ladder_rating)),
             str(int(player.ladder_rating_mean)),
-            str(int(player.ladder_rating_deviation))))
+            str(int(player.ladder_rating_deviation)))
 
         league = player.league
         if league is not None:
-            formatting = ("Division : {}\n"
-                          "Global Rating: {}")
-            self.rankItem.setToolTip(formatting.format(
-                league["division"],
-                str(int(rating))))
-            self.rankItem.setIcon(util.THEME.icon("chat/rank/%s.png" % league["league"]))
+            icon_str, tooltip_str = league["league"], "Division : " + league["division"] + "\n" + tooltip_str
         else:
-            self.rankItem.setIcon(util.THEME.icon("chat/rank/newplayer.png"))
-
-        self._verifySortOrder()
+            icon_str = "newplayer"
+        self.rankItem.setIcon(util.THEME.icon("chat/rank/%s.png" % icon_str))
+        self.rankItem.setToolTip(tooltip_str)
 
     def updateGame(self):
+        self.update_status_tooltip()
+        self.update_status_icon()
+        self.update_map()
+
+    def update_status_tooltip(self):
+        # Status tooltip handling
+        game = self.user_game
+        if game is not None and not game.closed():
+            private_str = " (private)" if game.password_protected else ""
+            if game.state == GameState.OPEN:
+                head_str = "Hosting" if game.host == self.user.name else "In Lobby"
```

Comment:
The head_str and add_str feel like something better expressed with formatting. We could do something like `header = "Hosting {private}game:"`, `header = In {private}lobby:`, `Playing{delay}:`, then just do `header.format(private=foo, delay=bar)` and use that for the second formatting.

I'm sorry for being so nitpicky, but I still have nightmares from refactoring the game tooltip string functions :( One day we're just going to use jinja2 for all this crap.

Reply:
come on you like it ;-)